### PR TITLE
feat(#724): New layer drawer and base layer selection

### LINKED ISSUE #724: feat: ability to switch to a satellite map
## Define your request
Currently there is no way to use a satellite map as a base layer.

**discord username: @el#9951**

## Define the alternatives you have in mind
Create a new base layer picker to switch between satellite and regular map.

## Extra context

Screenshot 2023-02-11 at 00 57 36

### PR DESCRIPTION
**Pull request by el**
# Description

Adds a satellite view switch to the UI

SCR-20230211-ktt

SCR-20230211-ksx

**discord username: el#9951**

**closes #724**

Please describe your changes. Also describe your aim and content. Do not forget to list the dependencies required caused by those changes.

## Things to check before creating a PR
- [ ] I have inspected my topic, checked.
- [ ] If it is a core feature, executed detailed tests.

## Creating PR rules
- [ ] PR must be created for an issue with approved tag. Otherwise PR will be rejected.
- [ ] Relevant issue number: The issue number related to PR must be attached to head of PR header, after prefix after prefix # must be attached in parenthesis. A header like this could be used "prefix(#issue_number): PR header" .
- [ ] A descriptive and understandable title: The PR title should clearly describe the nature and purpose of the changes. The PR title should be the first thing displayed when the PR is opened. And it should follow the semantic commit rules. For example, a title like "docs(#issueId): Add README.md" can be used.
- [ ] Related file selection: Only relevant files should be touched and no other files should be affected.
- [ ] Format and Lint Suitability : The code should be made in accordance with a certain format standard and examined according to the lint rules.
- [ ] Clean commit history: The commits where the changes are made should be clear and organized.
- [ ] Opening PR when job is completed: PR should be opened when job is completed and sent for review by other team members.

## Changes

- [ ] A new feature (a change that adds a new feature, not a breaking change)
- [ ] A new refactor (a change that is not a breaking change, that improves the readability or performance of the code)
- [ ] A breaking change
- [ ] Documentation change

# How were these changes tested?

Please describe the tests you did to test the changes you made. Please also specify your test configuration.

**Test Configuration**:

* Firmware version:
* Hardware:
* Toolchain:
* SDK:

### CODE REVIEW

File: pages/index.tsx [hunk 1]
```diff
@@ -190,14 +195,15 @@ export default function Home({ deviceType, singleItemDetail }: Props) {
                 }}
               >
                 <FilterMenu>
+                  <FilterMenu.TileLayer onChange={onTileChange} />
```
Comment (el) [inline pages/index.tsx:198]: I added the switch here but I am not sure if it is the best place for this.

File: components/UI/FilterMenu/TileLayer.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  return tileLayerMenuOptions.find(
+    (option) => option.value === value
+  ) as TileLayerMenuOption;
+};
+
+export interface TileLayerMenuProps {
+  onChange: (_option: TileLayerMenuOption) => void;
+}
+
+const TileLayerMenu: React.FC<TileLayerMenuProps> = ({ onChange }) => {
```
Comment (el) [inline components/UI/FilterMenu/TileLayer.tsx:34]: This component has a lot of shared logic with the other buttons in the filter menu, we should ideally refactor this.

File: public/locales/en/home.json [hunk 3]
```diff
@@ -42,6 +42,10 @@
         "noticeCount": "{{count}} notice available"
     },
     "map": {
+        "base": {
+            "map": "Map",
```
Comment (el) [inline public/locales/en/home.json:46]: Google and Yandex call these layers "Map/Satellite" while Bing calls them "Road/Aerial".

I think "Map/Satelite" is the better choice.

**Comment by eraygundogmus** [comment]
invalid pr.

**Comment by eraygundogmus** [comment]
we made some changes on UI, so your design is not compatible. therefore, you should not open a pr when there is UI/UX label on issue and there is not a figma file that we provided unless PO auth. thank you for your effort and understanding. please stand still till figma file become prepared

**Comment by eraygundogmus** [comment]
@el

**Comment by el** [comment]
@eraygundogmus Can you please tag me once it is available?

**Comment by eraygundogmus** [comment]
> @eraygundogmus Can you please tag me once it is available?

after a while of thinking, i found a better way to handle this. i assigned 'UI/UX' label to this issue. a pm will reach you out when its done. pr reopened to not discomfort you :)
apologize

**Comment by cdagli** [comment]
@el 👋

**Comment by canerakin111** [comment]
@el Hi El, we have a big community that has many experts in terms of design. We really appreciate all the effort. Since everyone would like to see the best outcome, it is important for us to get UX/UI team's opinion and output. We passed along your design too. We will keep you in the loop!

**Comment by sevilayerkan** [comment]
As test team we had tested this pr, new additions seems perfect to us we didn't see any problems for them. Congrats, this addition seems beautiful 👏

**Comment by el** [comment]
I talked with the design and I have recreated the PR with a new design from the UI/UX team. Will push it shortly

**Comment by el** [comment]
@sevilayerkan Can you test the PR again, we have changed the design completely.

File: components/UI/Drawer/components/LayerContent.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            checked={mapLayers.includes(MapLayer.Markers)}
+            title={t("map.layer.markers")}
+          />
+          <LayerButton
+            onClick={() => toggleMapLayer(MapLayer.Heatmap)}
+            image="heatmap"
+            checked={mapLayers.includes(MapLayer.Heatmap)}
+            title={t("map.layer.heatmap")}
+          />
+          {/* <LayerButton onClick={() => toggleMapLayer(MapLayer.Earthquakes)} image="bubblemap" checked={mapLayers.includes(MapLayer.Earthquakes)} title={t("map.layer.earthquakes")} /> */}
```
Comment (cdagli) [inline components/UI/Drawer/components/LayerContent.tsx:94]: Do you want to keep this or remove?

**Comment by cdagli** [comment]
It could be nice to preserve to selected map layer in the URL but with another issue maybe 👍

**Review by cdagli — APPROVED** [review]
LGTM 🌟

**Comment by sevilayerkan** [comment]
@el We retested it looks awesome we really liked the new feature. Congrats  🥳

**Comment by sevilayerkan** [comment]
@el We retested.There is a small issue regarding this PR : 
Steps to reproduce that : 
1 - Go to afetharita.com
2 - Zoom in 
3 - Open Map navigation on left
4 - Change Map Type from sth to sth else

Expected Result : The map should be visible properly
Actual Result : The map seem blank

Here is screen shot regarding to that:
image

**Review by oguzergul — APPROVED** [review]

File: components/UI/Drawer/components/LayerContent.tsx (hunk 4 above)
Comment (el) [reply components/UI/Drawer/components/LayerContent.tsx:94]: Keep it. Will work on it next

**Pull request merged**